camera_control/src/camera_control/camera_control_app.py

```diff
@@ -0,0 +1,114 @@
+import argparse
+import cv2
+import sys
+from PySide6.QtWidgets import (
+    QApplication,
+    QLabel,
+    QMainWindow,
+    QSlider,
+    QTabWidget,
+    QWidget,
+    QHBoxLayout,
+    QVBoxLayout,
+)
+from PySide6.QtCore import Qt
+from PySide6.QtGui import QImage, QPixmap
+
+from camera_control.camera_control_signals import CameraControlSignals
+from camera_control.detection_control_widget import DetectionControlWidget, DetectionParameters
+
+class CameraControlApp(QMainWindow):
+    def __init__(self, video_path):
+        super().__init__()
+        self.video_path = video_path
+
+        self.app_signals = CameraControlSignals()
+
+        self.setWindowTitle("Animal finder")
+        self.setGeometry(100, 100, 800, 600)
+
+        self.central_widget = QWidget(self)
+        self.setCentralWidget(self.central_widget)
+
+        # This label will display the video frames
+        self.frame_display = QLabel(self)
+        self.detection_mask_display = QLabel(self)
+
+        self.seek_slider = QSlider(Qt.Horizontal, self)
+        self.seek_slider.setMinimum(0)
+        self.seek_slider.setMaximum(100)
+        self.seek_slider.setValue(0)
+        self.seek_slider.valueChanged.connect(self.seek_frame)
+
+        default_detection_parameters = DetectionParameters(10, 11, "MOG2", 0.001)
+        self.detection_control_widget = DetectionControlWidget(self.app_signals, self.video_path, default_detection_parameters)
+
+        self.app_signals.frame_updated.connect(
+            lambda q_img: self.frame_display.setPixmap(QPixmap.fromImage(q_img))
+        )
+        self.app_signals.detection_completed.connect(
+            lambda q_img: self.frame_display.setPixmap(QPixmap.fromImage(q_img))
+        )
+        self.app_signals.detection_mask_updated.connect(
+            lambda q_img: self.detection_mask_display.setPixmap(QPixmap.fromImage(q_img))
+        )
+        self.app_signals.frame_number_updated.connect(self.detection_control_widget.set_frame_number)
+
+        self.running = True
+                
+        self.cap = cv2.VideoCapture(self.video_path)
+        if not self.cap.isOpened():
+            raise ValueError(f"Failed to open video: {self.video_path}")
+        self.total_frames = int(self.cap.get(cv2.CAP_PROP_FRAME_COUNT))
+
+        self.layout = QHBoxLayout(self.central_widget)
```

Comment:
I get an error here as well, because layout is a method ([code](https://github.com/python-qt-tools/PySide6-stubs/blob/30a43a39c4eda735003097444247876ee18408b9/PySide6-stubs/QtWidgets.pyi#L9253)).  Maybe in the version you are using there was no method?  I fixed it by renaming it to main_layout

Reply:
The method exists in my version too, but since it's python it happily allows setting reference to anything :-P